src/Highlighter.ts

```diff
@@ -171,4 +194,21 @@ export default class Highlighter {
       }
     }
   }
+
+  private compareRanges(range1: Range | null, range2: Range | null): boolean {
```

Comment:
what are you trying to accomplish here? can you use the native range comparison functions?

Reply:
just compare ranges
I've replaced last lines with:
```
    return range1!.compareBoundaryPoints(Range.START_TO_START, range2!) === 0
      && range1!.compareBoundaryPoints(Range.END_TO_END, range2!) === 0;
      ```